tests/test_java.py

```diff
@@ -718,3 +718,42 @@ def test_java_frames_include_no_semicolons(
         assert not is_pattern_in_collapsed(r"\([^);]+;[^)]*\)", collapsed)
         # only a pipe can occur within arguments
         assert is_pattern_in_collapsed(r"\([^);|]+\|[^)]*\)", collapsed)
+
+
+# test that async profiler doesn't print anything to applications stdout, stderr streams
+@pytest.mark.parametrize("in_container", [True])
+def test_no_stray_output_in_stdout_stderr(
+    tmp_path: Path,
+    application_pid: int,
+    application_docker_container: Container,
+    monkeypatch: MonkeyPatch,
+) -> None:
+    # save original stop function
+    stop_async_profiler = AsyncProfiledProcess.stop_async_profiler
+
+    # replace async profiler stop routine to trigger flushing standard streams
+    def flush_output_and_stop_async_profiler(self: AsyncProfiledProcess, *args, **kwargs) -> str:
+        # Call 'version' and 'check' actions on async-profiler to make sure writes to stdout, stderr are flushed
```

Comment:
Why does it cause flushing?

Anyway - a simpler solution: run the container, do a single profile, then stop the container; I suppose Java handles SIGTERM gracefully, so you will get the logs properly.

Reply:
That's true - java applications flush stdout by default (they trigger autoflush by default).  
async-profiler on the other hand is c++, and doesn't flush by default after each write to output stream.

I have observed that `version` action does trigger flush on standard output: ([profiler.cpp#L1548](https://github.com/granulate/async-profiler/blob/master/src/profiler.cpp#L1548)):
```cpp
        case ACTION_VERSION:
            out << PROFILER_VERSION;
            out.flush();
            break;
```
(this code is still there in the upstream, btw).

Reply:
I see. Good point. Please keep the code doing the `version` call and elaborate in a comment on why it's needed (you can also give a github permalink to this code which it the clear explanation why we want to run `version`).

Reply:
👍